snapcraft/plugins/plainbox_provider.py

```diff
@@ -46,3 +49,8 @@ def build(self):
             "python3", "manage.py", "install", "--layout=relocatable",
             "--prefix=/providers/{}".format(self.name),
             "--root={}".format(self.installdir)])
+
+        # Fix all shebangs to use the in-snap python.
+        file_utils.replace_in_file(self.installdir, re.compile(r''),
+                                   re.compile(r'#!.*python'),
+                                   r'#!/usr/bin/env python')
```

Comment:
Looks good, but it's missing a unit test. More than one when you implement the regexp suggestion.